File: 0724-find-pivot-index/724.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        left_sum, right_sum = 0, 0

        for num in nums:
            right_sum += num

        for i in range(0, len(nums)):
            right_sum -= nums[i]
            if left_sum == right_sum:
                return i
            left_sum += nums[i]

        return -1

File: 0724-find-pivot-index/724.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let (mut left_sum, mut right_sum) : (i32, i32) = (0, nums.iter().sum());

        for (i, &num) in nums.iter().enumerate() {
            right_sum -= num;
            if left_sum == right_sum {
                return i as i32
            }
            left_sum += num;
        }
        -1
    }
}
